tests/nextflowtower/test_utils.py

```diff
@@ -0,0 +1,191 @@
+                "preRunScript": "NXF_OPTS='-Xms4g -Xmx12g'",
+                "postRunScript": None,
+            },
+            "status": "AVAILABLE",
+            "orgId": 98765,
+            "workspaceId": 65748,
+        }
+    }
+
+    def test_get_compute_env(self, mocker, tower_utils):
+        mocked_request = mocker.patch.object(utils.TowerClient, "request")
+        mocked_request.return_value = self.EG_COMPUTE_ENV
+        compute_env_id = self.EG_COMPUTE_ENV["computeEnv"]["id"]
+        result = tower_utils.get_compute_env(compute_env_id)
+        assert result == self.EG_COMPUTE_ENV["computeEnv"]
+
+    EG_WORKFLOW = {
+        "workflow": {
+            "id": "7g2R5Z1J",
+            "runName": "tiny_shaw",
+            "sessionId": "n8b7v6-o4i5-u6q8w6-e2s7l1",
+            "profile": "test",
+            "workDir": "s3://test-project-tower-scratch/work",
+            "userName": "bgrande",
+            "revision": "3.9",
+            "commandLine": "nextflow run ...",
+            "projectName": "nf-core/rnaseq",
+            "launchId": "y1u2i3",
+            "status": "SUBMITTED",
+        },
+        "progress": {
+            "workflowProgress": {},
+            "processesProgress": [],
+        },
+        "platform": {"id": "aws-batch", "name": "Amazon Batch"},
+        "jobInfo": {},
+    }
+
+    def test_get_workflow(self, mocker, tower_utils):
+        mocked_request = mocker.patch.object(utils.TowerClient, "request")
+        mocked_request.return_value = self.EG_WORKFLOW
+        workflow_id = self.EG_WORKFLOW["workflow"]["id"]
+        result = tower_utils.get_workflow(workflow_id)
+        assert result == self.EG_WORKFLOW
+
+    EG_DICT = {
+        "foo": [1, 2, 3],
+        "bar": None,
+        "baz": {"tic": 1.1, "tac": 2.2, "toe": 3.3},
+    }
+
+    def test_update_dict_none(self, tower_utils):
+        # Expecting no change due to None value
+        overrides = {"foo": None}
+        result = tower_utils.update_dict(self.EG_DICT, overrides)
+        assert result == self.EG_DICT
+
+    def test_update_dict_invalid(self, tower_utils):
+        # Expecting no change due to None value
+        overrides = {"oof": "gah"}
+        with pytest.raises(ValueError):
+            tower_utils.update_dict(self.EG_DICT, overrides)
+
+    def test_update_dict_toplevel(self, tower_utils):
+        # Expecting top-level changes
+        overrides = {"foo": [4, 5, 6], "bar": False}
+        result = tower_utils.update_dict(self.EG_DICT, overrides)
+        assert result["foo"] == overrides["foo"]
+        assert result["bar"] == overrides["bar"]
+
+    def test_update_dict_nested(self, tower_utils):
+        # Expecting nested changes
+        overrides = {"baz": {"tic": 7.7, "tac": {"some": "thing"}}}
+        result = tower_utils.update_dict(self.EG_DICT, overrides)
+        assert result["baz"]["tic"] == overrides["baz"]["tic"]
+        assert result["baz"]["tac"] == overrides["baz"]["tac"]
+
```

Comment:
I think we should add this test to ensure that the original value under the `toe` key isn't overwritten.

```suggestion
        assert result["baz"]["toe"] == 3.3

```